pkg/registry/extensions/deployment/strategy.go

```diff
@@ -77,6 +77,14 @@ func (deploymentStrategy) PrepareForUpdate(ctx genericapirequest.Context, obj, o
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Status = oldDeployment.Status
 
+	// update is not allowed to set Spec.Selector for v1beta2
+	// we assume it is not v1beta2 if no API group and/or version is found in RequestInfo
+	// TODO(#50791): after v1beta1 is retired, move selector immutability check to validation codes
+	requestInfo, found := genericapirequest.RequestInfoFrom(ctx)
```

Comment:
@erictune @kow3ns As discussed by person, i added a TODO for moving the check to validation codes after _v1beta1_ retires, and included the API group in the condition.